src/port1.0/tests/portstartupitem.test

```diff
@@ -20,25 +20,81 @@ macports_worker_init
 package require port 1.0
 package require portstartupitem 1.0
 
-test startupitems {
-    startupitems unit test. Check that the name seen in foreach_startupitem\
-    is the same whether set via the startupitem.name option or via key+value\
-    in the startupitems option.
+
+test startupitems_executable {
+    Check that setting the startupitem.executable option behaves the\
+    same as setting executable via the startupitems option.
+} -setup {
+    set system_options(startupitem_install) yes
+} -body {
+    startupitem.executable  ${prefix}/sbin/mydaemon -flag1 "quoted string"
+    portstartupitem::foreach_startupitem {
+        set single_result [list {*}$si_executable]
```

Comment:
Why pull apart and then reassemble the list?

Reply:
It converts each list to its canonical form. I got that from [this tcl wiki page](https://wiki.tcl-lang.org/page/listcomp+-Compare+the+Contents+of+two+Lists) on the topic of how to compare lists for equality. Without that, the test failed because in one instance the quoted string was seen as being quoted with double quotes and in the other case was seen as being quoted with curly braces.